### ISSUE TITLE
why map pixels between `input_range`?

### ISSUE THREAD

**Issue by fcakyon** [bes-dev/pytorch_clip_guided_loss#1]
@bes-dev thanks for the awesome work! I have one question:

Why do you manually map the image pixels between -1 and 1, instead of directly using the https://huggingface.co/docs/transformers/model_doc/clip#transformers.CLIPFeatureExtractor?

**Comment by bes-dev** [comment]
@fcakyon thanks for your feedback!

So, we don't use transformers.CLIPFeatureExtractor because it isn't differentiable operation (they use PIL.Image transforms to make resize, etc.). As you can see from [VQGAN-CLIP sample](https://github.com/bes-dev/pytorch_clip_guided_loss/blob/master/examples/vqgan_clip/run.py), we can make backpropagation to input image. To do it, we use our differentiable implementation of the mapping input image to valid input tensor of the visual feature extractor.

**Comment by fcakyon** [comment]
@bes-dev makes perfect sense! Then this implementation is necessary for `vqgan` example and not essential for `cliprcnn` example, right?

**Comment by bes-dev** [comment]
@fcakyon yes, we use CLIP guided loss inference only for ranking without backward path in ClipRCNN.
